src/main.jsx

```diff
@@ -61,7 +62,20 @@ function App() {
       <GalleryNav direction={0}>⬅️</GalleryNav>
       <GalleryNav direction={1}>➡️</GalleryNav>
 
+      {/* Legacy Pagination Example
+       * <GalleryPagination renderPaginationItem={({ i }) => <span>{i + 1}</span>} />
+       */}
       <GalleryPagination renderPaginationItem={({ i }) => <span>{i + 1}</span>} />
+
+      <GalleryPagination>
+        {({ index, active }) => {
```

Comment:
I think we need to re-factor this. `GalleryMain` uses a prop `renderGalleryItem` to render its children and I think we should follow that pattern and use the prop we already have `renderPaginationItem`.

Reply:
This was sort of my point in the other comment. I don't know that we need to deprecate `renderPaginationItem`. In a way, I think I prefer that explicit implementation, rather than passing state to the `children()` function.

Reply:
My bad, interpreted it all wrong, I agree 💯

Reply:
@andrewrubin @marlonmarcello 
I think that my thought process is that components usually expect a children, in this case, children should always be a `GalleryPaginationItem` though it doesn't need to be that specific component.

I guess it just feels nicer if you could just write
```jsx
<GalleryPagination>
  <GalleryPaginationItem>Text</GalleryPaginationItem>
</GalleryPagination>
```

over 
```jsx
<GalleryPagination
    renderPaginationItem={() => <GalleryPaginationItem>Text</GalleryPaginationItem>}
/>
```

But I do realize we will usually want the props exposed, and it's more common to have

```jsx
<GalleryPagination
  renderPaginationItem={({ index, active }) => (
    <GalleryPaginationItem index={index} active={active} key={index}>
      <span>{index + 1}</span>
    </GalleryPaginationItem>
  )}
/>
```

over 
```jsx
<GalleryPagination>
  {({ index, active }) =>
    <GalleryPaginationItem index={index} active={active}>
      <span>{index + 1}</span>
    </GalleryPaginationItem>
  }}
</GalleryPagination>
```

Reply:
Thank you for the explanation, @Tiendongle  ❤️ 

I was leaning towards the explicit render prop simply because that is how the `GalleryMain` component works, and wanted to avoid a refactor of that to keep things consistent. I also have a small gripe with documenting the arguments passed to `children()` in the JSDoc style — it's a bit messy, in my opinion.

My vote is to leave children untouched, but I totally acknowledge that it is an accepted design pattern.

Reply:
I'm terribly unfamiliar with documenting in JSDoc style. So I hope it's okay there. I've moving things towards the renderProps to keep the consistency

Reply:
Doing this:
```jsx
<GalleryPagination>
  <GalleryPaginationItem>Text</GalleryPaginationItem>
</GalleryPagination>
```
Would also require the dev to get the `active` index from somewhere else like you acknowledge and mentioned:
> But I do realize we will usually want the props exposed

To be honest.. I also like to just use children directly like you mentioned and we can DEFINITELY achieve that, and I think we SHOULD. We already expose a hook for the user, we can just make that a requirement instead of passing render props and functions.

I mentioned this change because the pattern you were introducing was not in line with what we had already, but if we were to change both of them then that would awesome.

I think you should do that @Tiendongle probably in a new PR though

Reply:
With those updates we can achieve something like this:
```jsx
import { GALLERY_ITEMS } from "./some-data"
import { Gallery, GalleryMain, GalleryNav, GalleryPagination, GalleryItem } from "@wethegit/react-gallery"

const YourGallery = () => {
  const { activeIndex } = useGallery({
    totalItems: GALLERY_ITEMS.length,
    initialIndex: 0
  });

  return (
    <Gallery>
      <GalleryMain>
        {GALLERY_ITEMS.map((item, index) => (
          <GalleryItem key={index} active={activeIndex}>
            <img src={item.image} alt={item.alt} />
          </GalleryItem>
        )}
      </GalleryMain>

      <GalleryNav direction={0}>⬅️</GalleryNav>
      <GalleryNav direction={1}>➡️</GalleryNav>

      <GalleryPagination>
        {GALLERY_ITEMS.map((item, index) => (
          <GalleryPaginationItem key={index} active={activeIndex}>
            {index}
          </GalleryPaginationItem>
        )}
      </GalleryPagination>
    </Gallery>
  )
}

export default YourGallery
```

Reply:
I'm going to resolve this thread since we discussed it on Slack after the fact